torch/lib/THD/master_worker/worker/dispatch/Tensor.cpp

```diff
@@ -17,7 +17,75 @@ static std::unique_ptr<thpp::Tensor> createTensor(thpp::Type type) {
   throw std::invalid_argument("passed character doesn't represent a tensor type");
 }
 
-static void tensorConstruct(rpc::RPCMessage& raw_message) {
+static std::unique_ptr<thpp::Tensor> createTensor(thpp::Type type,
+                                                  THLongStorage *size,
+                                                  THLongStorage *stride) {
+  if (type == thpp::Type::UCHAR)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<unsigned char>(size, stride));
+  else if (type == thpp::Type::CHAR)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<char>(size, stride));
+  else if (type == thpp::Type::SHORT)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<short>(size, stride));
+  else if (type == thpp::Type::INT)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<int>(size, stride));
+  else if (type == thpp::Type::LONG)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<long>(size, stride));
+  else if (type == thpp::Type::FLOAT)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<float>(size, stride));
+  else if (type == thpp::Type::DOUBLE)
+    return std::unique_ptr<thpp::Tensor>(new thpp::THTensor<double>(size, stride));
+  throw std::invalid_argument("passed character doesn't represent a tensor type");
+}
+
+static std::unique_ptr<thpp::Tensor> createTensor(thpp::Type type,
```

Comment:
I think it would be easier to use a variadic template to this function, and use `std::forward` to pass the args to the tensor constructor.